src/components/Dialogs/AddLinkTdDialog.jsx

```diff
@@ -0,0 +1,294 @@
+        const opts = {
+          types: [
+            {
+              description: "Thing Description",
+              accept: { "application/ld+json": [".jsonld", ".json"] },
+            },
+          ],
+        };
+        if ("showOpenFilePicker" in window) {
+          return window.showOpenFilePicker(opts).then((handles) => handles[0]);
+        }
+        return window.chooseFileSystemEntries();
+      };
+
+     const getFileHTML5 = async () => {
+        return new Promise((resolve, reject) => {
+          const fileInput = document.getElementById("fileInput");
+          fileInput.onchange = (e) => {
+            const file = fileInput.files[0];
+            if (file) {
+              return resolve(file);
+            }
+            return reject(new Error("AbortError"));
+          };
+          fileInput.click();
+        });
+      };
+
+
+     const hasNativeFS = useCallback(() => {
+        return (
+          "chooseFileSystemEntries" in window || "showOpenFilePicker" in window
+        );
+      }, []);
+
+      /**
+       *
+       * @param {*} file
+       *
+       * Read file content
+       */
+
+      const linkingMethodChange= (linkingOption) => {
+          setlinkingMethod(linkingOption)
+          if(currentLinkedTd && linkingOption === 'url'){
+            setCurrentLinkedTd({})
+          }
+      }
+
+     const openFile = useCallback(
+        async (_) => {
+
+          if (!hasNativeFS()) {
+            const file = await getFileHTML5();
+            if (file) {
+                document.getElementById("link-href").value="./"+file.name
+            }
+            return;
+          }
+
+          let fileHandle;
+          try {
+            fileHandle = await getFileHandle();
+            const file = await fileHandle.getFile();
+            const href="./"+file.name
+            setCurrentLinkedTd(fileHandle)
+            document.getElementById("link-href").value=href
+          } catch (ex) {
+            console.log(ex)
+            const msg = "We ran into an error trying to open your TD.";
+            console.error(msg, ex);
+            alert(msg);
+          }
+        },
+        [hasNativeFS]
+      );
+
+     const children = <>
+         <label className="text-sm text-gray-400 font-medium pl-3">Thing Description:</label>
+         <div className="p-1">
+                {offlineTD["title"]}
+         </div>
+         <div className="p-1 pt-2">
+             <label htmlFor="rel" className="text-sm text-gray-400 font-medium pl-2">Relation:</label>
+             <input
+                list="relationType"
+                 type="text"
+                 name="rel"
+                 id="rel"
+                 className="border-gray-600 bg-gray-600 w-full p-2 sm:text-sm border-2 text-white rounded-md focus:outline-none focus:border-blue-500"
+                 placeholder="relation name"
+             />
+              <datalist id="relationType">
+                <option value="icon" />
```

Comment:
Can we have these options as array? 
Maybe a good idea could be to get them from the JSONschema or somewhere directly?